test/app_automate_test.rb

```diff
@@ -0,0 +1,279 @@
+require 'test_helper'
+require_relative '../lib/features/support/app_automate_driver'
+require_relative '../lib/features/support/capabilities/devices'
+
+class AppAutomateTest < Test::Unit::TestCase
+
+  USERNAME = "Username"
+  ACCESS_KEY = "Access_key"
+  LOCAL_ID = "Local_id"
+  APP_LOCATION = "App_location"
+
+  def test_default_assignments
+    driver = AppAutomateDriver.new(USERNAME, ACCESS_KEY, LOCAL_ID)
+    assert_equal(USERNAME, driver.instance_variable_get( :@username ))
```

Comment:
Ideally, you would try to avoid using `instance_variable_get` and `send` if you can. Instead, you can test the `@username`, `@access_key`, etc. were used correctly in certain places (like `#start_driver`).